test/mocha/001-schemas.js

```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
+      const extend = {name: 'test'};
+      // eslint-disable-next-line max-len
+      const schema = require('../node_modules/bedrock-validation/schemas/title')(extend);
+      const result = validation.validateInstance('Test Title', schema);
+      schema.name.should.equal('test');
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('url', function() {
+    const schema = validation.getSchema('url');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid url', function() {
+      const result = validation.validate('url', 'http://foo.com/v2');
+      result.valid.should.be.true;
+    });
+    it('should reject an invalid url', function() {
+      const result = validation.validate('url', {});
+      result.valid.should.be.false;
+    });
+    it('should accept valid url with extend', function() {
+      const extend = {name: 'test'};
+      // eslint-disable-next-line max-len
+      const schema = require('../node_modules/bedrock-validation/schemas/url')(extend);
+      const result = validation.validateInstance('http://foo.com/v2', schema);
+      schema.name.should.equal('test');
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('w3cDateTime', function() {
+    const schema = validation.getSchema('w3cDateTime');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid w3cDateTime', function() {
+      const result = validation.validate('w3cDateTime', '2016-01-01T01:00:00Z');
+      result.valid.should.be.true;
+    });
+    it('should reject an invalid w3cDateTime', function() {
+      const result = validation.validate('w3cDateTime', {});
+      result.valid.should.be.false;
+    });
+    it('should accept valid w3cDateTime with extend', function() {
+      const extend = {name: 'test'};
+      // eslint-disable-next-line max-len
+      const schema = require('../node_modules/bedrock-validation/schemas/w3cDateTime')(extend);
+      const result = validation.validateInstance(
+        '2016-01-01T01:00:00Z', schema);
+      schema.name.should.equal('test');
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('personName', function() {
+    const schema = validation.getSchema('personName');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid personName', function() {
+      const result = validation.validate('personName', 'Name');
+      result.valid.should.be.true;
+    });
+    it('should reject an invalid personName', function() {
+      const result = validation.validate('personName', {});
+      result.valid.should.be.false;
+    });
+    it('should accept valid personName with extend', function() {
+      const extend = {name: 'test'};
+      // eslint-disable-next-line max-len
+      const schema = require('../node_modules/bedrock-validation/schemas/personName')(extend);
```

Comment:
Not sure if this approach is going to survive moving to ESM. Is there another way? If not -- ok for now, we'll deal with ESM later.

Reply:
Well there probably is but there were also many tests already in this library which is why I used this structure. I think it's best we change that when we work on the ESM conversion.